src/s3fs.cpp

```diff
@@ -688,7 +688,7 @@ static FdEntity* get_local_fent(AutoFdEntity& autoent, const char* path, bool is
     }
 
     // open
-    time_t mtime         = (!S_ISREG(stobj.st_mode) || S_ISLNK(stobj.st_mode)) ? -1 : stobj.st_mtime;
+    time_t mtime         = (!S_ISREG(stobj.st_mode) && !S_ISLNK(stobj.st_mode)) ? -1 : stobj.st_mtime;
```

Comment:
Can we exercise this with a test somehow, maybe by limiting the statcache size?

Reply:
This bug is not related to stats and stats cache.
The stats information(meta) is used the correct value.
However, it sets mtime to -1 in the get_local_fent function and does not read the contents of the symlink file.
( https://github.com/s3fs-fuse/s3fs-fuse/blob/master/src/fdcache_entity.cpp#L482-L490 )
This is the part that prevents the contents of a file from being read except for a normal file, like an object that indicates a directory.
In the case of symlink, it is necessary to read it, but there was a problem with the judgment statement of mtime.